packages/ui/src/forum/components/Thread/WatchlistButton.tsx

```diff
@@ -11,9 +11,18 @@ interface Props {
 
 export const WatchlistButton = ({ threadId }: Props) => {
   const [watchlist, setWatchlist] = useLocalStorage<string[]>(FORUM_WATCHLIST)
-  const addToWatchlist = useCallback(() => setWatchlist(watchlist ? [...watchlist, threadId] : [threadId]), [watchlist])
+  const addToWatchlist = useCallback(
+    (e) => {
+      e.stopPropagation()
+      setWatchlist(watchlist ? [...watchlist, threadId] : [threadId])
+    },
+    [watchlist]
+  )
   const removeFromWatchlist = useCallback(
-    () => setWatchlist(watchlist?.filter((id) => id !== threadId) ?? []),
+    (e) => {
+      e.stopPropagation()
+      setWatchlist(watchlist?.filter((id) => id !== threadId) ?? [])
```

Comment:
Once #3719 will have made its way into `dev`. To prevent users from loosing some of their watchlist by adding threads from different browser tabs, it would be good to change these into something like:
```ts
setWatchlist((watchlist = []) => [...watchlist, threadId])
setWatchlist((watchlist = []) => watchlist.filter((id) => id !== threadId))
```